packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -67,8 +67,15 @@ export class WorkspaceFrontendContribution implements CommandContribution, MenuC
                     const rootNode = DirNode.createRoot(parentStat, name, label);
                     const dialog = this.fileDialogFactory({ title: WorkspaceCommands.OPEN.label! });
                     dialog.model.navigateTo(rootNode);
-                    const node = await dialog.open();
-                    this.openFile(node);
+                    const result = await dialog.open();
+                    if (result instanceof Array) {
```

Comment:
We should find a way to get rid it of it. all type checking is redundant here

Reply:
I added `FileSelection` return type and wrote a corresponding namespace with set of functions like `UriSelection`.